program/src/instruction.rs

```diff
@@ -1,184 +1,420 @@
+        desc = "Vault token account"
+    )]
+    #[account(
+        3,
+        name = "holder_rewards",
+        desc = "Holder rewards account for vault token account"
+    )]
+    #[account(
+        4,
+        writable,
+        name = "destination", 
+        desc = "Destination account for withdrawn lamports"
+    )]
+    #[account(
+        5,
+        signer,
+        name = "stake_authority",
+        desc = "Stake authority"
+    )]
+    #[account(
+        6,
+        name = "vault_authority",
+        desc = "Vault authority"
+    )]
+    #[account(
+        7,
+        name = "mint",
+        desc="Stake token mint"
+    )]
+    #[account(
+        8,
+        name = "spl_token_program",
+        desc = "SPL Token program"
+    )]
     HarvestHolderRewards,
 
     /// Harvests stake SOL rewards earned by the given stake account.
-    ///
+    /// 
     /// NOTE: This is very similar to the logic in the rewards program. Since the
     /// staking rewards are held in a separate account, they must be distributed
     /// based on the proportion of total stake.
-    ///
-    /// 0. `[w]` Config account
-    /// 1. `[w]` Stake account
-    /// 2. `[w]` Destination account
-    /// 3. `[s]` Stake authority
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Stake account"
+    )]
+    #[account(
+        2,
+        writable,
+        name = "destination",
+        desc = "Destination account for withdrawn lamports"
+    )]
+    #[account(
+        3,
+        signer,
+        name = "stake_authority",
+        desc = "Stake authority"
+    )]
     HarvestStakeRewards,
 
-    /// Slashes a stake account for the given amount
-    ///
+    /// Slashes a stake account for the given amount.
+    /// 
     /// Burns the given amount of tokens from the vault account, and reduces the
     /// amount in the stake account.
-    ///
-    /// 0. `[w]` Config account
-    /// 1. `[w]` Stake account
-    /// 2. `[s]` Slash authority
-    /// 3. `[w]` Vault token account
-    /// 4. `[]` Vault authority, PDA with seeds `['token-owner', stake_config]`
-    /// 5. `[]` SPL Token program
-    ///
-    /// Instruction data: amount of tokens to slash
+    /// 
+    /// Instruction data: amount of tokens to slash.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Stake account"
+    )]
+    #[account(
+        2,
+        signer,
+        name = "slash_authority",
+        desc = "Stake account"
+    )]
+    #[account(
+        3,
+        writable,
+        name = "vault",
+        desc = "Vault token account"
+    )]
+    #[account(
+        4,
+        name = "vault_authority",
+        desc = "Vault authority"
+    )]
```

Comment:
This is my own lack of knowledge with shank -- does it allow you to specify PDA seeds anywhere? Either way, could we add that info here?